#163422800 Implement Create Party Endpoint

haywhyze opened the pull request:
#### What does this PR do?
Implement Create Party Endpoint

#### Description of Task to be completed?
- write tests for the implementation of the create-party endpoint
- write a controller class to handle the routes `POST /parties`
- write a helper function to join words in a string
- write middleware to handle empty inputs
- write middleware to handle short or long party name
- write a data model for the party entity
- update test suites to accommodate file upload
- add middleware to upload party logo from client to server file system.
- add middleware to upload the logo to the cloud via cloudinary API
- add middleware to validate party headquarters address on Google Maps Places API
- add the middlewares to the router

#### How should this be manually tested?
- clone this repository by executing `git clone https://github.com/haywhyze/politico.git`
- cd to the cloned local repository `cd politico`
- run `ft-create-party-163422800`
- install all dependencies by running `npm install`
- start the server with `npm start`
- when the server is up, you can test using Postman or any other related service.

#### What are the relevant pivotal tracker stories?
[#163422800](https://www.pivotaltracker.com/story/show/163422800)

haywhyze committed:
feat(server): implement create party endpoint

- write test for the implementation of create party enpoint
- write a controller class to handle the routes `parties`
- write helper function to join words in a string
- write middleware to handle empty inputs
- write middleware to handle short or long party name
- write a data model for the party entity

haywhyze committed:
feat(server): implement create party endpoint

- update test suites to accomodate file upload
- add middleware to upload party logo from client to server file system.
- add middleware to upload the logo to the cloud via cloudinary API
- add middleware to validate party headquarters address on Google Maps  Places API
- add the middlewares to router

[delivers #163422800]

megame24 commented on server/middlewares/uploadLogoToCloud.js:
```diff
@@ -0,0 +1,31 @@
+const cloudinary = require('cloudinary').v2;
+
+const uploadLogoToCloud = (req, res, next) => {
+  // console.log('jjjjjjjjjjjjjjjj');
```
Get rid of console log and commented code

haywhyze replied:
This middleware has been merged with the uploadLogo middleware and the commented code deleted. 
As discussed on Friday, the .env.sample has been added but not on this branch. 
I added it together with some other configuration changes on [another branch](https://github.com/haywhyze/politico/commit/9b4b5aba5941140330b4e1766ec06ff1358382c1) to deploy the app to heroku to aid remote access to the app. 
I'll raise a PR for that as soon as this has been merged. 
Most recent PR feature ready for review will be live [here](https://politico-yusuf.herokuapp.com/api/v1/) so it can be tested on postman without having to clone and add api keys.

haywhyze committed:
feat(server): implement create party endpoint

- merge uploadLogo and uploadToCloud middleware
- handdled trimming of input from both address and name fields

[delivers #163422800]

megame24 reviewed (approved):
you can merge
